### PR TITLE
Functions to support REST authentication

### PR DESCRIPTION
**Pull request by gunnargestsson**
SHA, 
JwtSecurityToken,
UTC date time,
Xml <-> Json

### CODE REVIEW

**Comment by gunnargestsson** [comment]
I have signed cla :)

**Comment by gunnargestsson** [comment]
Hi

You should already have my agreement on file. Sent to Stuart in June.

Kveðja/Kind regards,
Gunnar Þór Gestsson
________________________________

File: DotNet Libraries.txt [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+      TimeZoneInfo := TimeZoneInfo.FindSystemTimeZoneById(TimeZone);
+      Duration := TimeZoneInfo.BaseUtcOffset;
+    END;
+
+    BEGIN
+    END.
+  }
+}
+
+OBJECT Codeunit 55986 Json/Xml Conversion Library
```
Comment (BardurKnudsen) [inline DotNet Libraries.txt]: Did you look at codeunit 1237 which does much of the same?

Comment (gunnargestsson) [reply DotNet Libraries.txt]: Yes, it does but only from json to xml.  I need both directions.  Having these functions will allow me to use XMLPorts for json files - until XMLPorts become DataPorts that will support json natively.

Comment (gunnargestsson) [reply DotNet Libraries.txt]: Example here: http://www.dynamics.is/?p=2637

**Comment by StanislawStempin** [comment]
Hi Gunnar, Even though you sent the agreement to Stuart, I would kindly ask you to sign the one requested by the GitHub bot. Otherwise all your future submissions will continue getting flagged as CLA-Required since this is an automated system as opposed to manual approval for the other private repository.

**Comment by gunnargestsson** [comment]
Stan, have accepted the CLA.  Waiting for the confirmation email.

File: DotNet Libraries.txt [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+    BEGIN
+      EXIT(EvaluateStandardDateTimeString(DateTimeString,'R'));
+    END;
+
+    PROCEDURE GetUTCDateTime@5() : DateTime;
+    VAR
+      DateTime@1000 : DotNet "'mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'.System.DateTime";
+    BEGIN
+      DateTime := DateTime.Now;
+      EXIT(DateTime)
```
Comment (Gallimathias) [inline DotNet Libraries.txt]: Allocation is unnecessary

EXIT(DateTime.Now);

Comment (gunnargestsson) [reply DotNet Libraries.txt]: Should be EXIT(DateTime.UTCNow);

**Comment by JesperSchulz** [comment]
Thank you very much for your contribution! We have now added the following functions to the following codeunits:

COD10 - Type Helper:
- TryEvaluateDateTime through Evaluate function
- EvaluateUTCDateTime
- FormatDateTime
- GetCurrUTCDateTime
- GetCurrUTCDateTimeAsText

COD1290 - SOAP Web Service Request Mgt.:
- HasJWTExpired

COD5459 - JSON Management:
- XMLTextToJSONText
- JSONTextToXMLText

TAB99008535 - TempBlob:
- GetImageType
- GetXMLAsText

We hope we actually managed to include all the useful functions you provided. Going forward, would it be possible to split your contributions in smaller logical pull requests? That makes it easier for us to ensure we get it all. Thank you very much again. If we missed anything or got something wrong, please create a corrective pull request on the submitted code. Keep the good functions comming :-)

**Comment by JesperSchulz** [comment]
The new library functions will ship with our November update. Thank you again!

**Pull request closed without merging**